lib/linter/linter.js

```diff
@@ -1283,6 +1286,7 @@ class Linter {
      */
     _verifyWithProcessor(textOrSourceCode, config, options, configForRecursive) {
         const filename = options.filename || "<input>";
+        const physicalFilename = filename;
```

Comment:
Source file can go through multiple processors (e.g. `file.md` -> `file.md/0_0.html` -> `file.md/0_0.html/0_0.js`), so at this point `options.filename` can already contain code block information.

Reply:
Can we expect `options.physicalFilename` to be present if this is second time through? 

```suggestion
        const filename = options.filename || "<input>";
        const physicalFilename = options.physicalFilename ?? filename;
```

Reply:
That is a good point. I think you’re correct.

Reply:
I think that would work. Also, we can use `filenameToExpose` variable (which is declared right below this line) instead of `filename`, to get the normalized version.

```js
const physicalFilename = options.physicalFilename || filenameToExpose;
```

Reply:
Updated 👍🏻